### ISSUE TITLE
Browser Ballot

### ISSUE THREAD

**Issue by bh-e** [UbuntuBudgie/budgie-welcome#142]
Where is browser ballot code reside? When I tried to run it in debian, Its says no installation candidate  for chromium-browser etc. I would like to tweak to run in my debian system.

**Comment by fossfreedom** [comment]
@bh-e you'll see that each html page has a click event that passes a code.

That code is looked up here https://github.com/UbuntuBudgie/budgie-welcome/blob/master/data/config/packages.json

So for each software item that could be installed, the details of how to install that are described in packages.json

**Comment by bh-e** [comment]
@fossfreedom, Thank you :)

**Comment by bh-e** [comment]
@fossfreedom, One more thing, If I want to an event, lets say I want to add midori browser. I changed gettingstarted.html  ( browser ballot  ) and package.json file. But when I run there is no Install/remove button.

**Comment by fossfreedom** [comment]
give me a weblink to your gettingstarted.html and packages.json so that I can have a quick look

**Comment by bh-e** [comment]
https://github.com/bh-e/budgie-welcome-deb

**Comment by fossfreedom** [comment]
I don't see where midori is in getting-started.html

the "<a id="firefox-install" and <"a id="firefox-remove" etc is important - the "-install" and "-remove" are the bits the code looks for to display or hide those buttons depending upon whether the package installed.

So "firefox" is the id in packages.json

I note you have changed the second block to "install" libreoffice - you need to change the app name  in "< id=libreoffice" and add libreoffice to packages.json



```
      <div class="col-md-6 application">
            <h4>&zwnj;Firefox&zwnj;</h4>
            <img src="img/applications/firefox.svg"/>
            <p class="text-md-justify">&zwnj;Mozilla Firefox is a free and open-source web browser, developed
            by the Mozilla Foundation - a not for profit organisation.  It is highly popular targeting multiple platforms.&zwnj;</p>
            <a id="firefox-install" class="btn btn-success" onclick="cmd('install?firefox');cmd('')">
              <span class="material-icons">&#xE2C0;</span><span class="button-text">&zwnj;Install&zwnj;</span>
            </a>
            <a id="firefox-remove" class="btn btn-danger" onclick="cmd('remove?firefox');cmd('')">
              <span class="material-icons">&#xE872;</span><span class="button-text">&zwnj;Remove&zwnj;</span>
            </a>
            <span id="firefox-status" class="hidden">
              <img src="img/welcome/processing.gif" style="width:24px" /> &zwnj;Applying Changes...&zwnj;
            </span>
          </div>
          <div class="col-md-6 application">
            <h4>&zwnj;Firefox&zwnj;</h4>
            <img src="img/applications/firefox.svg"/>
            <p class="text-md-justify">&zwnj;Mozilla Firefox is a free and open-source web browser, developed
            by the Mozilla Foundation - a not for profit organisation.  It is highly popular targeting multiple platforms.&zwnj;</p>
            <a id="firefox-install" class="btn btn-success" onclick="cmd('install?libreoffice');cmd('')">
              <span class="material-icons">&#xE2C0;</span><span class="button-text">&zwnj;Install&zwnj;</span>
            </a>
            <a id="firefox-remove" class="btn btn-danger" onclick="cmd('remove?libreoffice');cmd('')">
              <span class="material-icons">&#xE872;</span><span class="button-text">&zwnj;Remove&zwnj;</span>
            </a>
            <span id="firefox-status" class="hidden">
              <img src="img/welcome/processing.gif" style="width:24px" /> &zwnj;Applying Changes...&zwnj;
            </span>
          </div>
</div>
```


midoria would be like this

```
      <div class="col-md-6 application">
            <h4>&zwnj;Midori&zwnj;</h4>
            <img src="img/applications/firefox.svg"/>
            <p class="text-md-justify">&zwnj;Midoria is a free and open-source web browser, developed
            by the someone???? - a not for profit organisation.  It is highly popular targeting multiple platforms.&zwnj;</p>
            <a id="midori-install" class="btn btn-success" onclick="cmd('install?midori');cmd('')">
              <span class="material-icons">&#xE2C0;</span><span class="button-text">&zwnj;Install&zwnj;</span>
            </a>
            <a id="midori-remove" class="btn btn-danger" onclick="cmd('remove?midori');cmd('')">
              <span class="material-icons">&#xE872;</span><span class="button-text">&zwnj;Remove&zwnj;</span>
            </a>
            <span id="midori-status" class="hidden">
              <img src="img/welcome/processing.gif" style="width:24px" /> &zwnj;Applying Changes...&zwnj;
            </span>
          </div>
</div>
```

**Comment by bh-e** [comment]
Thanks, 
I've noticed another thing.
All html pages are shipped along with program. I've tried to change shop page by linking to the website. 
`<a class="btn btn-inverse fade fade-5s hvr-shadow" onclick="smoothPageFade('shop.html')">` 
to 
`<a class="btn btn-inverse fade fade-5s hvr-shadow" onclick="smoothPageFade('https://someshops.com')` 
But after first time caches are loaded and page changes doesn't reflect in webkit
